src/com/verisignlabs/dnssec/security/SignUtils.java

```diff
@@ -526,6 +526,15 @@ private static int ecdsaLength(int algorithm) throws SignatureException
     s_src_pos = (byte) (r_src_pos + r_src_len); s_pad = 0;
     len = (byte) (6 + r_src_len + s_src_len);
 
+    // leading zeroes are forbidden
+    if (signature[r_src_pos] == 0) {
```

Comment:
This should actually be a `while`, in case there's more than one leading zero. Will try to update soon.